syntax = "proto3";

package user;

service UserService {
  rpc CreateUser (CreateUserRequest) returns (CreateUserResponse);
  rpc GetUserById (GetUserByIdRequest) returns (GetUserByIdResponse);
}

message User {
  int64 id = 1;
  string username = 2;
  string email = 3;
}

message CreateUserRequest {
  string username = 1;
  string email = 2;
}

message CreateUserResponse {
  User user = 1;
}

message GetUserByIdRequest {
  int64 id = 1;
}

message GetUserByIdResponse {
  User user = 1;
}